packages/web-components/fast-components/src/radio/radio.ts

```diff
@@ -0,0 +1,212 @@
+
+    /**
+     * Tracks whether the "checked" property has been changed.
+     * This is necessary to provide consistent behavior with
+     * normal input radios
+     */
+    // TODO: marjon check if this is needed for radio buttons?
+    private dirtyChecked: boolean = false;
+
+    constructor() {
+        super();
+        this.proxy.setAttribute("type", "radio");
+    }
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+        this.updateForm();
+        if (this.name !== undefined) {
+            const radioGroup = radioGroups.get(this.name);
+            if (radioGroup) {
+                radioGroup.push(this);
+            } else {
+                radioGroups.set(this.name, [this]);
+            }
+        }
+
+        this.addEventListener("keydown", this.keydownHandler);
+    }
+
+    public disconnectedCallback(): void {
+        if (this.name !== undefined) {
+            const group = radioGroups.get(this.name);
+            if (group) {
+                group.splice(group.indexOf(this), 1);
+            }
+
+            if (group?.length === 0) {
+                radioGroups.delete(this.name);
+            }
+        }
+        this.removeEventListener("keydown", this.keydownHandler);
+    }
+
+    private updateForm(): void {
+        const value = this.checked ? this.value : null;
+        this.setFormValue(value, value);
+    }
+
+    public keydownHandler = (e: KeyboardEvent): void => {
+        const group = radioGroups.get(this.name);
+
+        switch (e.keyCode) {
+            case keyCodeArrowRight:
+            case keyCodeArrowUp:
+                if (this.name !== undefined) {
+                    if (group) {
+                        let index = group.indexOf(this) + 1;
+                        while (index < group.length) {
+                            if (!group[index].disabled) {
+                                group[index].checked = true;
+                                group[index].focus();
+                                break;
+                            } else {
+                                index += 1;
+                            }
+                        }
+                    }
+                }
+                break;
+            case keyCodeArrowLeft:
+            case keyCodeArrowDown:
+                if (this.name !== undefined) {
+                    if (group) {
+                        let index = group.indexOf(this) - 1;
+                        while (index >= 0) {
+                            if (!group[index].disabled) {
+                                group[index].checked = true;
+                                group[index].focus();
+                                break;
+                            } else {
+                                index -= 1;
+                            }
+                        }
+                    }
+                }
+                break;
+        }
+    };
```

Comment:
Should this belong in a radiogroup component? Would we expect other components to map to this API?

Reply:
using the name attribute to group radio button is the standard AFAIK. I started to create a radiogroup and got push back on whether it was necessary at all. I couldn't come up with a good reason to have the radiogroup component so this is a solution. If there are other components that have this standard of grouping by name attribute then maybe they should use this method as well? Hopefully I'm not misunderstanding your question here.

Reply:
Where did the pushback happen? I think I missed a conversation...

Reply:
@marjonlynch - one thing that we'll need to consider here is how these will work when they are within a toolbar: https://w3c.github.io/aria-practices/#for-radio-group-contained-in-a-toolbar

Does the current implementation support that behavior as outlined in the W3C specification or should we / will we need to add that?

Reply:
I had separate conversations with @nicholasrice and @EisenbergEffect around this topic and those conversations led me to this solution (the map was Rob's idea). Again, I started with the same idea you have, as you know I asked about doing the radio group. However given the way the <input type="radio"/> works out of the box this seemed like the better solution.  Actually I see this code could be made a few lines smaller as well now

Reply:
group stuff removed, just a humble radio left